File: own-cloud-service/back-end/java/spring/src/main/java/ua/edu/ontu/ocs/rest/api/v1_0_0/controller/user/AdminRestController.java
package ua.edu.ontu.ocs.rest.api.v1_0_0.controller.user;

import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import static ua.edu.ontu.ocs.rest.api.RestVersions.V1_0_0;;

@RestController
@RequestMapping(V1_0_0 + "/admin")
public class AdminRestController {

	@GetMapping("/login")
	public void login() {
		// TODO
	}

	@GetMapping("/logout")
	public void logout() {
		// TODO
	}

	@GetMapping("/read/all")
	public void readAll() {
		// TODO
	}

	@PostMapping("/create")
	public void create() {
		// TODO
	}

	@PutMapping("/update")
	public void update() {
		// TODO
	}

	@DeleteMapping("/delete")
	public void delete() {
		// TODO
	}
}

File: own-cloud-service/back-end/kotlin/reactive-spring/src/main/kotlin/ua/edu/ontu/ocs/rest/api/v1/endpoint/user/AdminRestController.kt
package ua.edu.ontu.ocs.rest.api.v1.endpoint.user

// TODO remove annotations
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping

@RequestMapping("/api/v1/admin")
class AdminRestController {

    @PostMapping("/sing-in")
    fun signIn() {
        // response
    }

    @GetMapping("/check-session")
    fun checkSession() {}

    @GetMapping("/sign-out")
    fun signOut() {}

    @PostMapping("/create")
    fun create() {
        // only localhost
    }
}